src/main/java/org/opensearch/flowframework/transport/PluginClient.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.transport;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionType;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.action.ActionResponse;
+import org.opensearch.identity.Subject;
+import org.opensearch.transport.client.Client;
+import org.opensearch.transport.client.FilterClient;
+
+/**
+ * A special client for executing transport actions as this plugin's system subject.
+ */
+public class PluginClient extends FilterClient {
+
+    private static final Logger logger = LogManager.getLogger(PluginClient.class);
+
+    private Subject subject;
+
+    public PluginClient(Client delegate) {
+        super(delegate);
+    }
+
+    public PluginClient(Client delegate, Subject subject) {
+        super(delegate);
+        this.subject = subject;
+    }
+
+    public void setSubject(Subject subject) {
+        this.subject = subject;
+    }
+
+    @Override
+    protected <Request extends ActionRequest, Response extends ActionResponse> void doExecute(
+        ActionType<Response> action,
+        Request request,
+        ActionListener<Response> listener
+    ) {
+        if (subject == null) {
```

Comment:
subject would be null with the first constructor call, right?

Reply:
we don't expect subject to be null, this is added to ensure robustness.